lib/Qandidate/Toggle/Serializer/ContextSerializer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Qandidate\Toggle\Serializer;
+
+use Qandidate\Toggle\Context;
+
+class ContextSerializer
+{
+    /**
+     * @param array $data
+     *
+     * @return Context
+     */
+    public function deserialize(array $data)
+    {
+        $this->assertHasKey('context', $data);
```

Comment:
I get your point, and I feel the same way for every serializer in the lib. But the truth is they are not actual serializers to me, they merely provide some useful method to flatten or reconstruct an object. But you're right, it's the control layer that should be responsible of such a behaviour, namely the API. Yet, I don't suppose you plan to remove all the serializers from the lib to make them available only in the API.